agent/command/papertrail_trace.go

```diff
@@ -0,0 +1,156 @@
+package command
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/evergreen-ci/evergreen/agent/internal"
+	"github.com/evergreen-ci/evergreen/agent/internal/client"
+	"github.com/evergreen-ci/evergreen/thirdparty"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+const (
+	productionPapertrailAddress = "https://papertrail.devprod-infra.prod.corp.mongodb.com"
+)
+
+type papertrailTrace struct {
+	Address   string   `mapstructure:"address" plugin:"expand"`
+	KeyID     string   `mapstructure:"key_id" plugin:"expand"`
+	SecretKey string   `mapstructure:"secret_key" plugin:"expand"`
+	Product   string   `mapstructure:"product" plugin:"expand"`
+	Version   string   `mapstructure:"version" plugin:"expand"`
+	Filenames []string `mapstructure:"filenames" plugin:"expand"`
+	WorkDir   string   `mapstructure:"work_dir" plugin:"expand"`
+	base
+}
+
+// This command is owned by the Dev Prod Release Infrastructure team
+func (t *papertrailTrace) Execute(ctx context.Context,
+	comm client.Communicator, logger client.LoggerProducer, conf *internal.TaskConfig) error {
+	if err := util.ExpandValues(t, &conf.Expansions); err != nil {
+		return errors.Wrap(err, "apply expansions")
+	}
+
+	pclient := &thirdparty.PapertrailClient{
+		Client:    &http.Client{},
+		Address:   t.Address,
+		KeyID:     t.KeyID,
+		SecretKey: t.SecretKey,
+	}
+
+	task := conf.Task
+
+	papertrailBuildID := fmt.Sprintf("%s_%d", task.Id, task.Execution)
```

Comment:
Since we do this in two places, could we just make a private function thats like "getPapertrailBuildID"?

Reply:
Done!